app/controllers/admin/recipients_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class Admin::RecipientsController < Admin::BaseController
+  load_and_authorize_resource :job_offer
+
+  def index
+    respond_to do |format|
+      format.json {
+        render json: recipients(params[:s]).to_json(
+          only: :id,
+          include: {
+            user: {
+              only: %i[first_name last_name email]
+            }
+          }
+        )
+      }
+    end
+  end
+
+  def create
```

Comment:
Is it normal that this is empty ?

Reply:
Oui c'est normal !

La magie du `load_and_authorize_resource` va instancier le `@job_offer` et un `@recipient` sur la base de ce qui est renvoyé par la méthode `recipient_params`.

Ensuite, l'action `create` rend la vue `create.js.erb`, qui se charge (grâce au `@job_offer` et au `@recipient`) de mettre à jour la liste de destinataires et le formulaire.